### PR TITLE
Refactor TraceAbstractionStarter

### PR DESCRIPTION
**Pull request by maul-esel**
Reasons for refactoring:

* properly support concurrent programs when analysing each error location separately
* code simplification
* preparation for future features e.g. analysing "insufficient thread" error locations separately

### CODE REVIEW

**Comment by maul-esel** [comment]
* `RegressionTestSuite` and `AutomizerRegressionTest` look good
* Travis CI fails because it also fails on `dev`

**Review by Heizmann — APPROVED** [review]

**Review by Heizmann — APPROVED** [review]

**Review by schuessf — APPROVED** [review]

File: trunk/source/TraceAbstraction/src/de/uni_freiburg/informatik/ultimate/plugins/generator/traceabstraction/TraceAbstractionStarter.java [hunk 1]
```diff
@@ -123,96 +137,55 @@ public TraceAbstractionStarter(final IUltimateServiceProvider services, final II
...
+		final IProgressMonitorService progmon = mServices.getProgressMonitorService();
+		mLogger.debug("Compute Hoare Annotation: " + mComputeHoareAnnotation);
 		mLogger.debug("Overall result: " + mOverallResult);
-		mLogger.debug("Continue processing: " + mServices.getProgressMonitorService().continueProcessing());
-		if (computeHoareAnnotation && mOverallResult != Result.TIMEOUT
-				&& !Result.USER_LIMIT_RESULTS.contains(mOverallResult)
-				&& mServices.getProgressMonitorService().continueProcessing()) {
+		mLogger.debug("Continue processing: " + progmon.continueProcessing());
+		if (mComputeHoareAnnotation && mOverallResult != Result.TIMEOUT
+				&& !Result.USER_LIMIT_RESULTS.contains(mOverallResult) && progmon.continueProcessing()) {
```
Comment (danieldietsch) [inline trunk/source/TraceAbstraction/src/de/uni_freiburg/informatik/ultimate/plugins/generator/traceabstraction/TraceAbstractionStarter.java]: it looks like the original intention here was at some point to debug the condition for continuing the analysis. 

I would just remove the `mLogger.debug` statements, but if you prefer to keep them add a fourth one for the missing conjunct

File: trunk/source/TraceAbstraction/src/de/uni_freiburg/informatik/ultimate/plugins/generator/traceabstraction/TraceAbstractionStarter.java [hunk 2]
```diff
@@ -123,96 +137,55 @@ public TraceAbstractionStarter(final IUltimateServiceProvider services, final II
...
 			final IBacktranslationService backTranslatorService = mServices.getBacktranslationService();
-			createInvariantResults(icfg, csToolkit, backTranslatorService);
+			createInvariantResults(icfg, icfg.getCfgSmtToolkit(), backTranslatorService);
```
Comment (danieldietsch) [inline trunk/source/TraceAbstraction/src/de/uni_freiburg/informatik/ultimate/plugins/generator/traceabstraction/TraceAbstractionStarter.java]: remove second parameter, extract inside method

**Review by danieldietsch — APPROVED** [review]
I like the PR very much, thank you! I noted some minor things, maybe you can address them and merge afterwards.

The reason that Travis is not building is because our Unit tests take too long. Ill try and get more time from them or disable the Travis build altogether.

**Comment by maul-esel** [comment]
Suggested changes made & merged.

**Pull request merged**